R/api_client.R

```diff
@@ -43,21 +55,21 @@ R6_orderlyweb_api_client <- R6::R6Class(
   "orderlyweb_api_client",
   cloneable = FALSE,
   public = list(
+    name = NULL,
     url = NULL,
     options = NULL,
     token = NULL,
     api_token = NULL,
 
-    initialize = function(hostname, port, token, https, prefix, api_version,
-                          insecure, verbose) {
-      self$url <- orderlyweb_api_client_url(hostname, port, https, prefix,
+    initialize = function(host, port, token, name, https, prefix,
+                          api_version, insecure, verbose) {
+      self$url <- orderlyweb_api_client_url(host, port, https, prefix,
                                             api_version)
-      ## If token is a function that should be ok too I think; that
-      ## would support the montagu flow well
-      if (!is.null(token)) {
-        assert_scalar_character(token)
-        self$token <- token
+      self$name <- orderlyweb_api_client_name(name, host, port, prefix)
+      if (!is.function(token)) {
```

Comment:
Is there any value in checking that, when token is a function,  it has the correct return type?

Reply:
there would be, but we can't do that without calling the function, and that breaks the desire for lazyness